internal/buildscripts/topotidy.py

```diff
@@ -0,0 +1,113 @@
+'''
+This Python script runs `go mod tidy` on all modules in the repository, in a way that guarantees
+convergence: no module should emit "updates to go.mod needed" errors, and running a second time
+should have no effect.
+
+This is achieved by running the command on modules in topological order, ie. running it on
```

Comment:
Consider having this list generated in a file so we can consult it, and update it via script?

Reply:
That's probably a good idea, although we'll probably need to implement yet another CI check to prevent the committed version from being out-of-date.

Reply:
No free lunch on that one

Reply:
What about outputting the list to a `.gitignore`d file?